source/_components/sensor.17track.markdown

```diff
@@ -0,0 +1,52 @@
+---
+layout: page
+title: "17track.net"
+description: "Instructions on how to use 17track.net data within Home Assistant"
+date: 2018-10-31 10:42
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: 17track.jpg
+ha_category: Postal Service
+ha_release: 0.82
+ha_iot_class: "Cloud Polling"
+---
+
+The 17track.net sensor platform allows users to get package data tied to their
+17track.net account. The platform creates both summary sensors, which show the
+number of packages in a current state (e.g., "In Transit"), as well as
+individual sensors for each package within the account.
+
+## {% linkable_title Configuration %}
+
+To enable the platform, add the following lines to your `configuration.yaml`
```

Comment:
Yes. The UI method was originally introduced so that users could add individual packages without needing to update `configuration.yaml`. Unfortunately, the method for tracking individual packages via 17track.net has been altered; until I figure out the mechanism, the underlying library (`py17track`) only allows account-driven package tracking.

Given that this use case is off the table, for now, there is one primary thing I review when considering whether UI config is appropriate: can the integration justify being a full-blown component (beyond the UI config use case)? Tactically:

1. Does the integration produce more than one platform (`sensor`, `binary_sensor`, etc.)?
2. Does the integration need to provide services?

If the answer is no, then I leave the integration as a platform (in this case, as a `sensor` platform). At some point, the HASS core will have a way for existing platforms to implement UI config without needing to transition to a full-blown component; when that occurs, I'll definitely take another look.

Hope that makes sense.

Reply:
Makes sense, thanks for explaining your decision!